### ISSUE TITLE
Linking error

### ISSUE THREAD

**Issue by patfelst** [sstaub/TickTwo#1]
Hi there, I'm trying to use this on M5Stack Core2 (ESP32). I'm using platformio. I don't think this library is registered in the arduino libraries, so I downloaded the zip, renamed folder to "TickTwo" and moved the folder to the platformio "inlcude" folder. It compiles ok however fails on linking:

```
Linking .pio/build/M5Unified/firmware.elf
.pio/build/M5Unified/src/main.cpp.o:(.literal._Z5setupv+0x30): undefined reference to `TickTwo::start()'
.pio/build/M5Unified/src/main.cpp.o:(.literal._Z4loopv+0x6c): undefined reference to `TickTwo::update()'
.pio/build/M5Unified/src/main.cpp.o:(.literal.startup._GLOBAL__sub_I_leds+0x10): undefined reference to `TickTwo::TickTwo(std::function<void ()>, unsigned int, unsigned int, resolution_t)'
.pio/build/M5Unified/src/main.cpp.o:(.literal.exit._GLOBAL__sub_D_leds+0x0): undefined reference to `TickTwo::~TickTwo()'
.pio/build/M5Unified/src/main.cpp.o: In function `setup()':
/Users/patrickfelstead/Documents/PlatformIO/Projects/M5Stack Core2 CO2 Monitor/.pio/libdeps/M5Unified/M5GFX/src/lgfx/v1/misc/colortype.hpp:585: undefined reference to `TickTwo::start()'
.pio/build/M5Unified/src/main.cpp.o: In function `loop()':
/Users/patrickfelstead/Documents/PlatformIO/Projects/M5Stack Core2 CO2 Monitor/.pio/libdeps/M5Unified/M5GFX/src/lgfx/v1/misc/colortype.hpp:585: undefined reference to `TickTwo::update()'
.pio/build/M5Unified/src/main.cpp.o: In function `_GLOBAL__sub_I_leds':
/Users/patrickfelstead/Documents/PlatformIO/Projects/M5Stack Core2 CO2 Monitor/.pio/libdeps/M5Unified/M5GFX/src/lgfx/v1/misc/colortype.hpp:585: undefined reference to `TickTwo::TickTwo(std::function<void ()>, unsigned int, unsigned int, resolution_t)'
.pio/build/M5Unified/src/main.cpp.o: In function `_GLOBAL__sub_D_leds':
/Users/patrickfelstead/Documents/PlatformIO/Projects/M5Stack Core2 CO2 Monitor/.pio/libdeps/M5Unified/M5GFX/src/lgfx/v1/misc/colortype.hpp:585: undefined reference to `TickTwo::~TickTwo()'
collect2: error: ld returned 1 exit status
*** [.pio/build/M5Unified/firmware.elf] Error 1
=========================================================================== [FAILED] Took 7.03 seconds ===========================================================================

Environment    Status    Duration
-------------  --------  ------------
M5Unified      FAILED    00:00:07.028
====================================================================== 1 failed, 0 succeeded in 00:00:07.028 ======================================================================
The terminal process "platformio 'run', '--environment', 'M5Unified'" terminated with exit code: 1.

Terminal will be reused by tasks, press any key to close it.
```

I'm probably doing something wrong, but not sure what. Do you have any suggestions please?

I can provide my code, but it uses the same format as your example.

**Comment by sstaub** [comment]
Did you try to rebuild intellisense?

**Comment by patfelst** [comment]
yes I did rebuild intellisense.

Interestingly, "Ticker.h" seems to work ok for me, i.e. there didn't seem to be any naming conflicts.

**Comment by sstaub** [comment]
Interesting, one reason for that library was that many people have problems with naming conflicts with the ESP platform.

**Comment by patfelst** [comment]
Actually, I will investigate Ticker some more. I have a suspicion I have something else included as PlatformIO autocomplete indicates I don't have .pause() and .resume() available. I will report back when I have time to investigate.

**Comment by patfelst** [comment]
Yes it seems the espressif arduino core also has a Ticker library (/.platformio/packages/framework-arduinoespressif32/libraries/Ticker/src/Ticker.h) which was being included as I hadn't installed Ticker from the library manager, hence your note about the reason for TickerTwo to avoid naming conflicts. Apoligies for this.

But I still have the linking error in TickTwo.

**Comment by sstaub** [comment]
Can you compile with the Arduino IDE?

**Comment by patfelst** [comment]
Solved. I moved the `TickerTwo` folder into the PIO `libdeps` folder and now it links. And it runs as expected too scheduling a function every 1.0 seconds.

Didn't link:
<img width="758" alt="Screen Shot 2021-11-15 at 9 16 42 pm" src="https://user-images.githubusercontent.com/11342773/141768657-45db5566-79a2-4ce9-9d6d-64fce2726ae7.png">

Links and runs ok:
<img width="758" alt="Screen Shot 2021-11-15 at 9 16 16 pm" src="https://user-images.githubusercontent.com/11342773/141768735-74a7c436-74ea-48f9-b027-502663de229b.png">
